File: Assignment_39/program3.cpp
#include<iostream>
using namespace std;

template <class T>
int SearchFirst(T *arr, int iSize, T iNo)
{
    int i = 0;

    for(i = 0; i < iSize; i++)
    {
        if(arr[i] == iNo)
        {
            break;
        }
    }

    if(i == iSize)
    {
        return -1;
    }
    else
    {
        return i + 1;
    }
}

int main()
{
    int arr[] = {10,20,30,10,40,10,40,10};

    int iRet = SearchFirst(arr,9,40);

    cout<<iRet;

    return 0;
}

File: Assignment_55/program3.java
import java.util.*;

class MyArray
{
    public char Arr[];

    public MyArray(int iSize)
    {
        Arr = new char[iSize];
    }

    public void Accept()
    {
        Scanner sobj = new Scanner(System.in);

        int iCnt = 0;

        System.out.println("Enter the array elements : ");
        for(iCnt = 0; iCnt < this.Arr.length; iCnt++)
        {
            this.Arr[iCnt] = sobj.nextLine().charAt(0);
        }
    }

    public void Display()
    {
        int iCnt = 0;

        System.out.println("Elements of the array are : ");
        for(iCnt = 0; iCnt < this.Arr.length; iCnt++)
        {
            System.out.print(this.Arr[iCnt]+"\t");
        }
        System.out.println();
    }

    public int Search()
    {
        Scanner sobj = new Scanner(System.in);

        int iCount = 0, i = 0;
        char ch = '\0';
    
        System.out.println("Enter the character that you want to search : ");
        ch = sobj.nextLine().charAt(0);

        for(i = 0; i < Arr.length; i++)
        {
            if((Arr[i] == ch) || (Arr[i] == ch - 32) || (Arr[i] == ch + 32))
            {
                iCount++;
            }
        }

        return iCount;
    }
}

class program3
{
    public static void main(String A[])
    {
        Scanner sobj = new Scanner(System.in);
        int iRet = 0;
        char str = '\0';

        System.out.println("Enter the number of characters : ");
        int iSize = sobj.nextInt();

        MyArray mobj = new MyArray(iSize);

        mobj.Accept();

        mobj.Display();

        iRet = mobj.Search();  

        System.out.println("Frequency of that character is : "+iRet);
    }
}